### PR TITLE
Support for complete token refurbishment

### PR DESCRIPTION
**Pull request by 8845musign**
# Changes

- Support for complete token refurbishment
- I test by comparing the Storybook before and after applying the token on every page.

## 

# Check

- [ ] ~Browser verification (minimum) Android Chrome/iOS Safari(375px-)~ Almost only colour change
- [ ] ~CSS not affected by inheritance~
- [ ] ~Layout does not break even if there is an overflow~
- [ ] ~Layout does not break when wraps~
- [ ] ~Added new Component~
  - [ ] ~Added data-* prop and id prop~
- [x] Updated [Ubie Vitals](https://github.com/ubie-oss/ubie-vitals-website) or Added an update [issue](https://github.com/ubie-oss/ubie-vitals-website/issues)(if needed)

### CODE REVIEW

File: src/components/Box/Box.module.css [hunk 1]
```diff
@@ -12,53 +12,11 @@
   line-height: var(--text-leading);
   color: var(--text-color);
   overflow-wrap: anywhere;
+  background-color: var(--background-color);
```
Comment (8845musign) [inline src/components/Box/Box.module.css:15]: I changed from specifying variations by class to CSS Variables
In the future, if the number of variations increases or decreases, the number of changes will be reduced.

File: src/components/Button/Button.module.css [hunk 2]
```diff
@@ -32,7 +32,7 @@
 }
 
 .button:focus-visible {
-  outline: solid var(--color-accent) 2px;
+  outline: solid var(--focus-ring, var(--color-border-blue-darken)) 2px;
```
Comment (8845musign) [inline src/components/Button/Button.module.css:35]: In the focus-ring, pink is only used when the component color is blue. While blue is the basic color, it is used exceptionally as another color.

File: src/components/Divider/Divider.module.css [hunk 3]
```diff
@@ -4,13 +4,6 @@
...
-
-.divider.gray {
-  background-color: var(--color-border);
-}
-
-.divider.primary {
-  background-color: var(--color-primary);
-}
+  border: 0;
+  border-top: var(--border-width) solid var(--border-color);
```
Comment (8845musign) [inline src/components/Divider/Divider.module.css:8]: Change from class to CSS Variables

File: src/components/Flex/Flex.tsx [hunk 4]
```diff
@@ -55,7 +55,7 @@ type Props = {
        * 子要素の間隔。指定しない場合は0
        * xxs=4px, xs=8px, sm=12px, md=16px, lg=24px, xl=40px, xxl=64px
        */
-      spacing: Spacing;
+      spacing?: Spacing;
```
Comment (8845musign) [inline src/components/Flex/Flex.tsx:58]: When the spacing notation was supported, the absence of spacing was no longer acceptable.
I've changed it back to optional.

File: src/components/Heading/Heading.module.css [hunk 5]
```diff
@@ -84,6 +69,11 @@
   text-decoration: underline;
 }
 
+.heading.xxs {
```
Comment (8845musign) [inline src/components/Heading/Heading.module.css:72]: Add Size Variation

File: src/components/Heading/Heading.tsx [hunk 6]
```diff
@@ -1,33 +1,21 @@
...
 
 type Props = {
   /**
    * テキストの配置。指定しない場合、親要素の配置を継承
    */
   textAlign?: 'left' | 'center' | 'right';
-  /**
-   * アイコン。プライマラーカラーで表示。icon propはどれかひとつのみを指定してください
-   */
-  primaryIcon?: ReactNode;
```
Comment (8845musign) [inline src/components/Heading/Heading.tsx:16]: Deleted because the icon was not used.

File: src/components/Text/Text.module.css [hunk 7]
```diff
@@ -3,7 +3,7 @@
   font-weight: normal;
   hyphens: auto;
   line-height: var(--leading);
-  color: var(--color);
+  color: var(--text-color);
```
Comment (8845musign) [inline src/components/Text/Text.module.css:6]: In the CSS file, I was using variables to make the specification. However, this time I made the colour specification directly from JavaScript.

File: src/types/style.ts [hunk 8]
```diff
@@ -1,16 +1,28 @@
-export type FontSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
+export type FontSize = 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | 'xl';
```
Comment (8845musign) [inline src/types/style.ts:1]: This file defines the variations of most components.

File: src/utils/style.ts [hunk 9]
```diff
@@ -117,29 +123,61 @@ export const cssLeadingToken = ({
   return '';
 };
 
-export const colorVariable = (color: TextColor | undefined): CSSProperties => {
+const PascalToKebab = (str: string) => {
```
Comment (8845musign) [inline src/utils/style.ts:126]: The functions in this file convert the colour keys specified in the component to CSS Variables.

File: package.json [hunk 10]
```diff
@@ -127,5 +127,8 @@
   "dependencies": {
     "@react-docgen/cli": "^2.0.4",
     "debounce": "^2.1.1"
+  },
+  "optionalDependencies": {
```
Comment (8845musign) [inline package.json:131]: GitHub Actions fails to install this package. Even if it fails, the process can continue.

ref: https://github.com/npm/cli/issues/4828

**Review by takanorip — APPROVED** [review]

**Pull request merged**